### PR TITLE
Payoff Vector Refactor Completed

### LINKED ISSUE #11: Refactor cpp code to Julia
Refactor the payoff vector code (c++) into Julia so it can be fully compatible with the rest of our project.

### PR DESCRIPTION
**Pull request by Dan2k03**
Refactored version of the payoff vector code in last issue. It is now fully in Julia.

### CODE REVIEW

File: recsourcePayoutRefactor.jl [hunk 1]
```diff
@@ -0,0 +1,83 @@
+using Printf
+
+# Province struct
+struct Province
+    name::String
+    money::Int
+end
+
+# Function to generate the power set of provinces
+function power_set(country::Vector{Province})
```
Comment (GeorgeR227) [inline recsourcePayoutRefactor.jl]: We should be able to use Combinatorics.jl for this.

File: recsourcePayoutRefactor.jl [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            push!(pow_list, temp_vec)
+        end
+    end
+
+    # Remove the empty set
+    return pow_list[2:end]
+end
+
+# Function to print payoffs based on conversion rates
+function print_payoffs(power_set::Vector{Vector{Province}}, conversion_rates::Dict{Int,Int})
```
Comment (GeorgeR227) [inline recsourcePayoutRefactor.jl]: Let's split this into a function that computes/returns the payoffs and another that prints. I think current format for payoffs now is a `Dict` of the list of players (a coalition basically) to their value.

File: recsourcePayoutRefactor.jl [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            elseif cost < val_sum
+                if temp_closest < cost
+                    temp_closest = cost
+                    best_fit_crate = crates
+                end
+            end
+        end
+
+        # Print the result
+        println("(" * join(name_list, ", ") * ") can buy: $best_fit_crate crates with a total of \$$val_sum")
```
Comment (GeorgeR227) [inline recsourcePayoutRefactor.jl]: We can overload `Base.show` for our payoffs so we don't need to run a special function for that.

**Review by GeorgeR227 — CHANGES_REQUESTED** [review]

File: recsourcePayoutRefactor.jl (hunk 1 above)
Comment (Dan2k03) [reply recsourcePayoutRefactor.jl]: Done.

File: recsourcePayoutRefactor.jl (hunk 2 above)
Comment (Dan2k03) [reply recsourcePayoutRefactor.jl]: I just commented out the print function for simplicity's sake. I siphoned all the the results into a dictionary which will be used when I integrate your code with mine.

**Review by GeorgeR227 — APPROVED** [review]
Great, I'm going to go ahead and merge this.

**Pull request merged**